waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -107,9 +106,9 @@ generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
 generatePrismaClient genProjectRootDirAbs = do
   chan <- newChan
   (_, dbExitCode) <-
-    concurrently
-      (readJobMessagesAndPrintThemPrefixed chan)
-      (DbJobs.generatePrismaClient genProjectRootDirAbs chan)
+    readJobMessagesAndPrintThemPrefixed chan
```

Comment:
Could running two jobs concurrently with `readJobMessagesAndPrintThemPrefixed` be the source of some of that output confusion? The minute it gets one job exit message it stops printing.

Also, given we got an error in the output for one of my test runs, I wonder if we should just run them sequentially to play it safe?

Reply:
Yeah, I think this is only possible place where the bug could arise. @Martinsos and I had a discussion on whether we should go with two files (`schema.server.prisma` and `schema.client.prisma`) or do something else (e.g.,. creating symbolic links for either the files or the generated directory, edit the file in-memory and call the prisma with that, create a temporary file and delete it, etc.)...

But now, with all the output and concurrency confusion (we'll report the same generation messages twice), I think it might be best to ditch this approach and instead:
1. Generate the `.prisma/client` directory for the server
2. Copy it to the web app's node_modules

What do you think @shayneczyzewski (I know what @Martinsos thinks :))?

Reply:
Yeah, I'd be ok with doing that instead. My only concern is do we know for sure that `.prisma/client` is 100% standalone and relocatable? For instance, are there any other things we have to copy over, and are there any paths in there that are relative to server stuff? If they generate identical output, I think copying is fine. 👍🏻

Reply:
It should be, but I'll jump over to their discord and make sure.

Reply:
This is also what worries me. Even if it is now, will it be in the future? Hm.
But as I mentioned in another comment, if generation is quick, we can just do it twice but do it sequentially or hide one of the outputs.

Reply:
Just thinking about this more over lunch. There are a few other potential downsides with the copy approach:
- our `deps` command is kinda lying for the web app by not showing `@prisma/client` (plus, if they ever added it for some unknown reason, could that cause some headaches for us, or in the future when we start splitting our deps vs their deps, will this bite us)?
- does NPM care if it has extra, untracked folders in the `node_modules` dir? will it ever get silently cleaned/removed by any of our normal commands and we are left in a bad state?

If we want to keep going down this approach, which is probably "safer" and more semantically accurate, I think doing the following should help with the double output and the possible concurrency Prisma error of running generate twice simultaneously:
```hs
  (_, serverExitCode) <-
    concurrently (readJobMessagesAndPrintThemPrefixed chan) (DbJobs.generatePrismaClientForServer genProjectRootDirAbs chan)
  clientExitCode <- DbJobs.generatePrismaClientForClient genProjectRootDirAbs chan
```

Reply:
All good points, I'm ditching the plan B and going back to plan A :D
Thanks for the suggestion, I'll try it out and report here!

Reply:
Ok, I tried it out, and it indeed solves the concurrency problem, but there are several things I don't like:
- We only print the server messages, making the output server-specific. It seems arbitrary and unusual to suppress one of the outputs like this, so I vote for keeping both outputs. 
- It seems wrong that the database is the one that says this - it should come from the server and the client, respectively. It also might make sense for server and client directories to contain their respective Prisma schemas. I understand we wanted to keep the database stuff near the database. Regardless, I think it makes sense to move them.

Assuming we agree on generating the clients sequentially, a few questions remain unanswered.

I see several possible approaches:
1. Two schema files/clients generated, run the generation steps sequentially, show server output as coming `Db`, suppress client output (i.e., leave things as they currently are).
2. Two schema files/clients generated, run the generation steps sequentially, and show both outputs as coming from `Server` and `Web app` (not `Db`).
3. **NEW DISCOVERY:** One schema file, generate a _single shared client_ in `out/.node_modules` directory (the import resolution works automatically). There's only a _single generation step_ which we run as coming from `Db`.

I vote for option 3).

If we're going with options 1) or 2), we also have to decide between the following:
1. Keeping the schema files in `db`.
2. Keeping the schema files in `server` and `web-app`.

@Martinsos @shayneczyzewski What do you think?

Reply:
I'm a bit out of my depth on the intricacies of the resolution and the implications of this structure, but I would say as long as it works in the editor, when running locally, and when building/deploying, option 3 sounds good to me. :)

Would this need/produce a `package.json` in the `out/` root as well? I only ask because of that one bug we hit with Prisma (https://github.com/prisma/prisma/issues/11770), but I don't think that it would impact it.

Reply:
Same as @shayneczyzewski @sodic , if option 3) works in all cases and environemnts, then sure! It does sound like it might be tricky when deploying though, possibly. You will know that the best.
But if we need to go instead with other options, then I would say both 1) and 2) sound good. Maybe 2) a bit better.

Summary: I don't have to much offer here I think, sounds like you can @sodic make the best choice here, just make sure it works in all scenarios.

Reply:
Argh, deploying couldn't work (at least not in the way it currently works). I'll stick with option 2) for now

Reply:
@Martinsos What about the second question, do you have a preference on where we should keep the schema files?

Reply:
Hm not sure @sodic ! At this point, whatever you think is best, I think I will agree with.

Reply:
This is what we have in the end:
1. We generate a single schema file (`shema.prisma`)
2. We set the `client.output` field to gets its value from the environment variable `PRISMA_CLIENT_OUTPUT_DIR`
3. We generate two Prisma clients (for both the server and the web app) by passing a different value of the `PRISMA_CLIENT_OUTPUT_DIR` env var to the same code and schema.